### PR TITLE
add Test for running shared object macro

### PR DESCRIPTION
**Pull request by ferdymercury**
matches https://github.com/root-project/root/pull/14813

### CODE REVIEW

**Comment by dpiparo** [comment]
Thanks for this PR, this is great. We need to impose a dependency between the two tests though, as CTest will try to run them in parallel. We can express the dependency in CMake, or write a simple shell/python script that 
1. writes the hello.C macro to disk
2. Builds the library with aclic
3. runs the library as a macro
4. cleans up the space

This latter solution has the advantage of being self contained in 1 single simple file.
What do you think?

**Comment by ferdymercury** [comment]
> Thanks for this PR, this is great. We need to impose a dependency between the two tests though, as CTest will try to run them in parallel. We can express the dependency in CMake, or write a simple shell/python script that
> 
>     1. writes the hello.C macro to disk
> 
>     2. Builds the library with aclic
> 
>     3. runs the library as a macro
> 
>     4. cleans up the space
> 
> 
> This latter solution has the advantage of being self contained in 1 single simple file. What do you think?

Done. Not sure though whether the working_directory and outcnv variables should be like this or not.

**Comment by ferdymercury** [comment]
> Done. Not sure though whether the working_directory and outcnv variables should be like this or not.

Solved now!

**Review by dpiparo — APPROVED** [review]

File: root/meta/hello_ROOT.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#! /bin/sh
```
[the remark this answers is not in the document]
Comment (pcanal) [reply root/meta/hello_ROOT.sh:1]: Actually the feature pre-dates the use of FIXTURE and the example we have (see root/io/stdpair for example)
use `ROOTTEST_COMPILE_MACRO` and `${COMPILE_MACRO_TEST}`

**Review by pcanal — CHANGES_REQUESTED** [review]
Let's replace the shell script.

File: root/meta/hello_ROOT.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ferdymercury) [reply root/meta/hello_ROOT.sh]: Fixed as suggested

**Comment by ferdymercury** [comment]
It's not able to find the compiled macro. Is it stored in a different directory the result maybe?

**Comment by ferdymercury** [comment]
> It's not able to find the compiled macro. Is it stored in a different directory the result maybe?

Ok, I think it's solved now!

**Comment by guitargeek** [comment]
Thanks! For tests we don't necessarily need backports, by the way. As long as one branch tests the features it's good!

**Review by guitargeek — APPROVED** [review]

**Pull request merged**